### PR TITLE
task_DEV-3. Radix converter.

### PR DESCRIPTION
**Pull request by egorsolovyev98**

### CODE REVIEW

File: task_DEV-3/task_DEV-3/RadixConverter.cs [hunk 1]
```diff
@@ -0,0 +1,66 @@
...
+                '0' , '1' , '2' , '3' , '4' , '5' ,
+                '6' , '7' , '8' , '9' , 'a' , 'b' ,
+                'c' , 'd' , 'e' , 'f' , 'g' , 'h' ,
+                'i' , 'j' , 'k' , 'l' , 'm' , 'n' ,
+                'o' , 'p' , 'q' , 'r' , 's' , 't' ,
+                'u' , 'v' , 'w' , 'x' , 'y' , 'z'
+            };
+
+            char[] buf = new char[33];
+            bool negative = (value < 0);
```
Comment (IgorKheidorov) [inline task_DEV-3/task_DEV-3/RadixConverter.cs:43]: Magic number, comment is needed

File: task_DEV-3/task_DEV-3/RadixConverter.cs [hunk 2]
```diff
@@ -0,0 +1,66 @@
...
+            int charPos = 32;
+
+            if (!negative)
+            {
+                value = -value;
+            }
+
```
Comment (IgorKheidorov) [inline task_DEV-3/task_DEV-3/RadixConverter.cs:49]: value <= -radix, explain what do you mean by this

File: task_DEV-3/task_DEV-3/RadixConverter.cs [hunk 3]
```diff
@@ -0,0 +1,66 @@
...
+            char[] digits = 
+            {
+                '0' , '1' , '2' , '3' , '4' , '5' ,
+                '6' , '7' , '8' , '9' , 'a' , 'b' ,
+                'c' , 'd' , 'e' , 'f' , 'g' , 'h' ,
+                'i' , 'j' , 'k' , 'l' , 'm' , 'n' ,
+                'o' , 'p' , 'q' , 'r' , 's' , 't' ,
+                'u' , 'v' , 'w' , 'x' , 'y' , 'z'
+            };
+
```
Comment (IgorKheidorov) [inline task_DEV-3/task_DEV-3/RadixConverter.cs:40]: Why buf size is 33? explain

File: task_DEV-3/task_DEV-3/RadixConverter.cs [hunk 4]
```diff
@@ -0,0 +1,66 @@
...
+            char[] buf = new char[33];
+            bool negative = (value < 0);
+            int charPos = 32;
+
```
Comment (IgorKheidorov) [inline task_DEV-3/task_DEV-3/RadixConverter.cs:44]: Explain why do you make positive value to be negative

**Review by IgorKheidorov — CHANGES_REQUESTED** [review]

File: task_DEV-3/task_DEV-3/RadixConverter.cs (hunk 3 above)
Comment (egorsolovyev98) [reply task_DEV-3/task_DEV-3/RadixConverter.cs:40]: The maximum possible number of divisions with the remainder = 31 and 1 for sign

File: task_DEV-3/task_DEV-3/RadixConverter.cs (hunk 1 above)
Comment (egorsolovyev98) [reply task_DEV-3/task_DEV-3/RadixConverter.cs:43]: The maximum number of divisions with the remainder = 31 and is possible with the maximum value for the variable of the type int = 2147483647 and the minimum radix= 2.

File: task_DEV-3/task_DEV-3/RadixConverter.cs (hunk 4 above)
Comment (egorsolovyev98) [reply task_DEV-3/task_DEV-3/RadixConverter.cs:44]: The input number can be both negative and positive, therefore it is necessary to bring the radix and the number to the same sign, I decided to reduce both numbers to negative

File: task_DEV-3/task_DEV-3/RadixConverter.cs (hunk 2 above)
Comment (egorsolovyev98) [reply task_DEV-3/task_DEV-3/RadixConverter.cs:49]: If the number and radix are reduced to a negative sign, the cycle condition is changed to the above

**Pull request merged**